src/components/DownloadModal/index.tsx

```diff
@@ -349,9 +356,14 @@ export class DownloadModal extends Component<DownloadModalProps, DownloadModalSt
 
     return result;
   };
+
+  setPageCount = (totalPages: number) => {
+    this.setState({ pageCount: totalPages });
+  };
+
   confirmSelection() {
     const pdfDownLoadProps: PDFDownloadLinkRenderProps = {
-      document: createDocument({ ...this.state.submitDownloadProps }),
+      document: createDocument({ ...this.state.submitDownloadProps }, this.setPageCount),
```

Comment:
Good idea to pass in a callback! 👍

Reply:
I have fixed the test errors, however I can't seem to find a way to pull setPageCount out of the return statement because the totalPages argument I'm 'stealing' seems to only exist in the context of the <text> element's render attribute. I may be wrong/missing a way of getting those values.  The React-PDF documentation shows totalPages as an available argument in the render attribute here: https://react-pdf.org/advanced#dynamic-content